src/capture/header_middleware/header_middleware.h

```diff
@@ -26,50 +26,11 @@
 #ifndef HEADER_MIDDLEWARE_H
 #define HEADER_MIDDLEWARE_H
 
-#include <sqlite3.h>
-#include <pcap.h>
-#include <stdint.h>
-
-#include "../../utils/allocs.h"
-#include "../../utils/os.h"
-#include "../../utils/eloop.h"
-
-#include "packet_queue.h"
-#include "../pcap_service.h"
-
-/**
- * @brief Initialises the header middleware
- *
- * @param db The sqlite3 db
- * @param db_path The sqlite3 db path
- * @param eloop The eloop structure
- * @param pc The pcap context
- * @return struct middleware_context* the middleware context on success, NULL on
- * failure
- */
-struct middleware_context *init_header_middleware(sqlite3 *db, char *db_path,
-                                                  struct eloop_data *eloop,
-                                                  struct pcap_context *pc);
+#include "../middleware.h"
 
 /**
- * @brief Header packet processors
- *
- * @param context The middleware context
- * @param ltype The packet type
- * @param header The pcap packet header
- * @param packet The pcap packet
- * @param ifname The capture interface
- * @return int 0 on success, -1 on failure
+ * @brief Header Middleware.
+ * @author Alexandru Mereacre, Alois Klink
```

Comment:
We should add a description as to what the header middleware does. I'm guessing it stores the packet header data in the capture.sqlite database?

Reply:
Exactly. It stores the packet headers accordingly to the given schema.

Reply:
Documented in 703faf635ad3cbe868087f843f082bc942129e0f